src/transformers/models/bark/processing_bark.py

```diff
@@ -0,0 +1,295 @@
+        to the tokenizer and to `cached_file` method if `voice_preset` is a valid filename.
+
+        Args:
+            text (`str`, `List[str]`, `List[List[str]]`):
+                The sequence or batch of sequences to be encoded. Each sequence can be a string or a list of strings
+                (pretokenized string). If the sequences are provided as list of strings (pretokenized), you must set
+                `is_split_into_words=True` (to lift the ambiguity with a batch of sequences).
+            voice_preset (`str`, `Dict[np.ndarray]`):
+                The voice preset, i.e the speaker embeddings. It can either be a valid voice_preset name, e.g
+                `"en_speaker_1"`, or a directly a dictionnary of embeddings for each submodel of `Bark`. Or it can be a
+                valid file name of a local `.npz` single voice preset.
+            return_tensors (`str` or [`~utils.TensorType`], *optional*):
+                If set, will return tensors of a particular framework. Acceptable values are:
+
+                - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                - `'np'`: Return NumPy `np.ndarray` objects.
+
+        Returns:
+            Tuple([`BatchEncoding`], Dict[~utils.TensorType]): A tuple composed of a [`BatchEncoding`], i.e the output
+            of the `tokenizer` and a `Dict[~utils.TensorType`]`, i.e the voice preset with the right tensors type.
+        """
+        if voice_preset is None or isinstance(voice_preset, dict):
+            pass
+        else:
+            voice_preset_key = voice_preset.replace("/", "_")
+            if isinstance(voice_preset, str) and f"{voice_preset_key}_semantic_prompt" in self.speaker_embeddings_dict:
+                voice_preset = {}
+                for prompt_type in [
+                    "semantic_prompt",
+                    "coarse_prompt",
+                    "fine_prompt",
+                ]:
+                    voice_preset[prompt_type] = self.speaker_embeddings_dict[f"{voice_preset_key}_{prompt_type}"]
+
+            else:
+                if isinstance(voice_preset, str) and not voice_preset.endswith(".npz"):
+                    voice_preset = voice_preset + ".npz"
+
+                voice_preset = np.load(voice_preset)
+
+        self._validate_voice_preset_dict(voice_preset)
+        voice_preset = convert_dict_to_tensors(voice_preset, return_tensors)
+        encoded_text = self.tokenizer(
+            text,
+            return_tensors=return_tensors,
+            padding="max_length",
+            max_length=max_length,
+            return_attention_mask=return_attention_mask,
+            return_token_type_ids=return_token_type_ids,
+            add_special_tokens=add_special_tokens,
+            **kwargs,
+        )
+
+        return encoded_text, voice_preset
+
+
+def convert_dict_to_tensors(tensor_dict, tensor_type: Optional[Union[str, TensorType]] = None):
```

Comment:
We should try and avoid adding a new ad-hoc function like this to the processor -> it won't get updates that we make to the feature extractor / tokenizer's `convert_to_tensors` methods so risks getting outdated and a burden to maintain